File: Castro-Lara-Daniel_Ivan/Procesos/files C++/hilos.cpp
/*
    Sistemas Operativos
    Daniel Iván Castro Lara
    Programa en C++
	$ gcc -o procesos_v0_param procesos_v0_param.cpp
*/

#include <iostream>
#include <thread>
#include <unistd.h>  // Para getpid()

// Función para el hilo secundario
void thread() {
    while (true) {
        std::cout << "\t\t\t H=" << getpid() << std::endl;
    }
}

int main(int argc, char **argv) {
    try {
        // Crear el hilo
        std::thread t(thread);

        // Bucle infinito en el hilo principal
        while (true) {
            std::cout << "\tP=" << getpid() << std::endl;
        }

        // Esperar a que el hilo termine (no se ejecutará en este caso)
        t.join();
    } catch (const std::exception &e) {
        std::cerr << "Error en la creación del hilo: " << e.what() << std::endl;
        return 1;
    }

    return 0;
}

File: Castro-Lara-Daniel_Ivan/Procesos/files Python/hilos.py
"""
    Sistemas Operativos
    Daniel Iván Castro Lara
    Programa en Python 
"""

import threading
import os
import time

# Función para el hilo secundario
def thread():
    while True:
        print(f"\t\t\tH={os.getpid()}")
        time.sleep(0.5)  # Pausa para evitar un consumo excesivo de CPU

def main():
    try:
        # Crear el hilo
        thread_obj = threading.Thread(target=thread)
        thread_obj.start()
    except Exception as e:
        print("Error en la creación del hilo:", e)
        return

    # Bucle infinito en el hilo principal
    while True:
        print(f"\tP={os.getpid()}")
        time.sleep(0.5)  # Pausa para evitar un consumo excesivo de CPU

if __name__ == "__main__":
    main()

"""
    1.- python <archivo>.py
    No se necesita compilar porque es interpretado.
"""
